python/run_final_analysis.py

```diff
@@ -446,17 +488,27 @@ def run(rdf_module, args) -> None:
         # Now perform the loop and evaluate everything at once.
         LOGGER.info('Evaluating...')
         all_events_raw = dframe.Count().GetValue()
+
         LOGGER.info('Done')
 
         nevents_real += all_events_raw
         uncertainty = ROOT.Math.sqrt(all_events_raw)
 
         if do_scale:
             LOGGER.info('Scaling cut yields...')
-            all_events = all_events_raw * 1. * gen_sf * \
-                int_lumi / process_events[process_name]
-            uncertainty = ROOT.Math.sqrt(all_events_raw) * gen_sf * \
-                int_lumi / process_events[process_name]
+            if do_weighted:
+                #should add check if the weight column exists!
+                all_events_weighted = dframe.Sum("weight").GetValue() #this will only work if "weight" column is defined in the analysis script, should weight name be an argument?
```

Comment:
Can you do try/catch block? or always define "weight" row regardless.

Reply:
Done, works to catch the exception now, thanks!